README.md

```diff
@@ -50,14 +54,59 @@ To conveniently deploy a StatefulSet to OpenShift a [Template is provided](kafka
 
-If you want to use other connectors, you can:
-* Mount a volume containing the plugins to path `/opt/kafka/plugins/`
-* Use the `enmasseproject/kafka-connect` image as Docker base image, add your connectors to the `/opt/kafka/plugins/` directory and use this new image instead of `enmasseproject/kafka-connect`
-
 ### Deploying to OpenShift
 
 To conveniently deploy Kafka Connect to OpenShift a [Template is provided](kafka-connect/resources/openshift-template.yaml), this could be used via `oc create -f kafka-connect/resources/openshift-template.yaml` so it is present within your project. To create a whole Kafka Connect cluster use `oc new-app barnabas-connect` *(make sure you have already deployed a Kafka broker)*.
 
 ### Deploying to Kubernetes
 
 To conveniently deploy Kafka Connect to Kubernetes use the provided yaml files with the Kafka Connect [deployment](kafka-connect/resources/kafka-connect.yaml) and [service](kafka-connect/resources/kafka-connect-service.yaml). This could be done using `kubectl apply -f kafka-connect/resources/kafka-connect.yaml` and `kubectl apply -f kafka-connect/resources/kafka-connect-service.yaml`.
+
+### Using Kafka Connect with additional plugins
+
+Our Kafka Connect images contain by default only the `FileStreamSinkConnector` and `FileStreamSourceConnector` connectors.
+If you want to use other connectors, you can:
+* Mount a volume containing the plugins to path `/opt/kafka/plugins/`
+* Use the `enmasseproject/kafka-connect` image as Docker base image, add your connectors to the `/opt/kafka/plugins/` directory and use this new image instead of `enmasseproject/kafka-connect`
+* Use OpenShift build system and our S2I image
+
+#### Mount a volume containing the plugins
+
+* Prepare a persistent volume which contains a directory with your plugin(s).
+* Mount the volume into your Pod
+
+#### Create a new image based on `enmasseproject/kafka-connect`
+
+* Create new `Dockerfile` which uses `enmasseproject/kafka-connect`
+```Dockerfile
+FROM enmasseproject/kafka-connect:latest
+USER root:root
+COPY ./my-plugin/ /opt/kafka/plugins/
+USER kafka:kafka
+```
+* Build the Docker image and upload it to your Docker repository of choice
+* Use your new Docker image in your Kafka Connect deployment
+
+#### Using Openshift build and S2I image
+
+* Create OpenShift build configuration using our OpenShift template
+```
+oc apply -f kafka-connect/s2i/resources/openshift-template.yaml
+oc new-app barnabas-connect-s2i
+```
+* Prepare a directory with Kafka Connect plugins which you want to use
+* Start new image build using the prepared directory
+```
+oc start-build kafka-connect --from-dir ./my-plugins/
+```
+* Find out the address of your local Docker image repository (for example from the build logs `oc logs -f bc/kafka-connect`)
```

Comment:
Can we use a command substitution with a `grep` so that the the `oc new-app -p ...` command you introduced a few lines down "just works"?

Reply:
Hmm, tricky comment ... the output looks like this:
```
$ oc logs -f bc/kafka-connect
Receiving source from STDIN as archive ...
Assembling plugins into custom plugin directory /tmp/kafka-plugins

Pushing image 172.30.1.1:5000/myproject/kafka-connect:latest ...
Pushed 5/7 layers, 73% complete
Pushed 6/7 layers, 89% complete
Pushed 7/7 layers, 100% complete
Push successful
```

I can write a command which should work but it looks like this:
```
oc logs -f bc/kafka-connect | grep "Pushing image" | cut -d " " -f 3 | cut -d / -f 1,2
```
IMHO the current README makes it a bit more clear what to look for. If we use the command and someone has slightly different configuration and the command doesn't work it will be completely unclear what to use. The current example makes it at least a bit clear.

**WDYT?**

Reply:
A less cumbersome, and more easily understood pipeline would be something like `oc logs -f bc/kafka-connect | awk '/Pushing image (.*) .../{ print $3 }'`.

So long as you've explained what the purpose of the pipeline is, then if the command doesn't work it should be easy for them to fix it for themselves.

Reply:
@tombentley I updated the README.md to make it a bit more clear what the Build does. It includes the simplified command although slightly different from the one you suggested. Please have a look.